File: server/typescript/packages/render/test/recover/strip.test.ts
import { describe, test, expect } from "bun:test";
import { strip } from "../../src/recover/strip.js";

// Mirrors Java StripTest / C# StripTests (FR-010 stage 1).
describe("strip", () => {
  test("unwraps json fence", () => {
    const out = strip('Sure! Here you go:\n```json\n{"a":1}\n```\nHope that helps.');
    expect(out).toContain('{"a":1}');
    expect(out).not.toContain("```");
  });

  test("unwraps bare fence", () => {
    expect(strip('```\n{"a":1}\n```').trim()).toBe('{"a":1}');
  });

  test("unwraps xml fence", () => {
    const out = strip("```xml\n<a>1</a>\n```");
    expect(out).toContain("<a>1</a>");
    expect(out).not.toContain("```");
  });

  test("no fence returns trimmed input", () => {
    expect(strip('   {"a":1}   ')).toBe('{"a":1}');
  });

  test("null/undefined safe", () => {
    expect(strip(null)).toBe("");
    expect(strip(undefined)).toBe("");
  });
});

File: server/java/render/src/test/java/com/metaobjects/render/extract/StripTest.java
package com.metaobjects.render.extract;

import org.junit.Test;
import static org.junit.Assert.*;

public class StripTest {

    @Test
    public void unwrapsJsonFence() {
        String in = "Sure! Here you go:\n```json\n{\"a\":1}\n```\nHope that helps.";
        String out = Strip.strip(in);
        assertTrue(out.contains("{\"a\":1}"));
        assertFalse(out.contains("```"));
    }

    @Test
    public void unwrapsBareFence() {
        assertEquals("{\"a\":1}", Strip.strip("```\n{\"a\":1}\n```").trim());
    }

    @Test
    public void unwrapsXmlFence() {
        String out = Strip.strip("```xml\n<a>1</a>\n```");
        assertTrue(out.contains("<a>1</a>"));
        assertFalse(out.contains("```"));
    }

    @Test
    public void noFenceReturnsTrimmedInput() {
        assertEquals("{\"a\":1}", Strip.strip("   {\"a\":1}   "));
    }

    @Test
    public void nullSafe() {
        assertEquals("", Strip.strip(null));
    }
}
